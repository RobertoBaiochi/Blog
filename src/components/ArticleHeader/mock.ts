import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

// const { posts } = data;
// const {
//   id,
//   attributes: { title, excerpt, createdAt },
// } = posts.data[1];

// export default {
//   id,
//   title,
//   excerpt,
//   cover: data.posts.data[1].attributes.cover,
//   author: data.posts.data[1].attributes.author.data,
//   categories: data.posts.data[1].attributes.categories.data,
//   createdAt,
// } as unknown as ArticleHeaderProps;

const {
  id,
  attributes: { title, excerpt, createdAt },
} = data.posts.data[1];

export default {
  id,
  title,
  excerpt,
  cover: data.posts.data[1].attributes.cover,
  author: data.posts.data[1].attributes.author.data,
  categories: data.posts.data[1].attributes.categories.data[1],
  createdAt,
} as unknown as ArticleHeaderProps;
