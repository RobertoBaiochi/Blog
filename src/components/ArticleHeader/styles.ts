import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
