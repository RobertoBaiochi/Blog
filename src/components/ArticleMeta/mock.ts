import { ArticleMetaProps } from './';
import { data } from '../../api/data.json';

const { posts } = data;

const {
  attributes: {
    createdAt,
    author: {
      data: {
        id: idAuthor,
        attributes: { displayName: displayNameAuthor, slug: slugAuthor },
      },
    },
    categories,
  },
} = posts.data[1];

export default {
  createdAt,
  author: {
    idAuthor,
    displayNameAuthor,
    slugAuthor,
  },
  categories,
} as unknown as ArticleMetaProps;

// import { ArticleMetaProps } from '.';

// export default {
//   createdAt: '2023-05-09T04:49:38.694Z',
//   author: {
//     id: '1',
//     displayName: 'Roberto Baiochi',
//     slug: 'roberto-baiochi',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'ReactJs',
//       slug: 'react-js',
//     },
//     {
//       id: '2',
//       displayName: 'NextJs',
//       slug: 'next-js',
//     },
//   ],
// } as ArticleMetaProps;
