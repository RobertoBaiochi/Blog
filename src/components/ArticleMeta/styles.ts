import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
