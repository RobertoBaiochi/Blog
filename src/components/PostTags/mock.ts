import { PostTagsProps } from '.';
import { data } from '../../api/data.json';

export default {
  tags: data.posts.data[0].attributes.tags.data,
} as unknown as PostTagsProps;

// export default { tags: data.posts.data[0] } as unknown as PostTagsProps;
