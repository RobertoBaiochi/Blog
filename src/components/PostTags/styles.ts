import styled, { css } from 'styled-components';

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;

    span {
      margin-left: ${theme.spacing.xsmall};
      &::after {
        content: ', ';
      }

      &::before {
        content: ' ';
      }

      &:last-child::after {
        content: '';
      }
    }

    a {
      text-decoration: none;
      transition: all 300ms ease-in-out;

      @media( hover: hover) {
        &:hover {
          filter: brightness(50%);
        }
      }
    }

  `}
`;
