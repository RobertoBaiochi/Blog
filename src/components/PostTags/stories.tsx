import { Meta, StoryFn } from '@storybook/react';
import PostTags, { PostTagsProps } from '.';

import mock from './mock';

export default {
  title: 'PostTags',
  component: PostTags,
  args: mock,
} as Meta<PostTagsProps>;

export const Template: StoryFn<PostTagsProps> = (args) => {
  return (
    <div>
      <PostTags {...args} />
    </div>
  );
};

export const NoTags: StoryFn<PostTagsProps> = (args) => {
  return (
    <div>
      <PostTags {...args} tags={undefined} />
    </div>
  );
};
