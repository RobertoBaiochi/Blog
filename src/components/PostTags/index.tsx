import Link from 'next/link';
import { Tags } from '../../shared-typed/tags';
import { Wrapper } from './styles';

export type PostTagsProps = {
  tags?: Tags[];
};

const PostTags = ({ tags = [] }: PostTagsProps) => {
  if (tags.length === 0) {
    return null;
  }

  return (
    <Wrapper>
      tags:
      {tags.map((tag) => {
        const {
          id,
          attributes: { slug, displayName },
        } = tag;
        return (
          <span key={`${slug}+${id}`}>
            <Link href={`/tags/${slug}`}>{displayName}</Link>
          </span>
        );
      })}
    </Wrapper>
  );
};
export default PostTags;
