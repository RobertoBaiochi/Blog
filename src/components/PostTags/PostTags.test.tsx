import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import PostTags, { PostTagsProps } from '.';
import '@testing-library/jest-dom';

import mock from './mock';

const props: PostTagsProps = mock;

describe('<PostTags />', () => {
  it('should render a link', () => {
    renderTheme(<PostTags {...props} />);

    expect(screen.getByRole('link')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostTags {...props} />);

    expect(container).toMatchSnapshot();
  });
});
