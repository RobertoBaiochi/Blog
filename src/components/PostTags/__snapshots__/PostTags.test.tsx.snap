// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTags /> should match snapshot 1`] = `
<div>
  <p
    class="sc-40ba6cc8-0 jwQYTv"
  >
    tags:
    <span>
      <a
        href="/tags/react"
      >
        React
      </a>
    </span>
  </p>
</div>
`;

exports[`<PostTags /> should match snapshot with no tags 1`] = `<div />`;
